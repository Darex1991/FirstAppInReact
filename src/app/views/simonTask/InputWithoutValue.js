import React from 'react'
import { FormInput } from 'react-form'

export default function InputWithoutValue ({field, showErrors, errorBefore, isForm, noTouch, ...rest}) {
  return (
    <FormInput field={field} showErrors={showErrors} errorBefore={errorBefore} isForm={isForm}>
      {({setValue, getValue, setTouched}) => {
        return (
          <input
            {...rest}
            onBlur={() => setTouched(true)}
          />
        )
      }}
    </FormInput>
  )
}
