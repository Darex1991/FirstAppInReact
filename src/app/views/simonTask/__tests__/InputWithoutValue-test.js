import React            from 'react';
import ReactTestUtils   from 'react-addons-test-utils'

import chai             from 'chai';
import chaiJestSnapshot from "chai-jest-snapshot";

import InputWithoutValue from '../InputWithoutValue';

chai.use(chaiJestSnapshot);

test('Should render InputWithoutValue correctly', () => {
    const shallowRenderer = ReactTestUtils.createRenderer();
    const result = shallowRenderer.render( <InputWithoutValue /> );
    expect(result).toMatchSnapshot();
});
