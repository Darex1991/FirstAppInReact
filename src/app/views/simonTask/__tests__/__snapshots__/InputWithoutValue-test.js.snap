exports[`test Should render InputWithoutValue correctly 1`] = `
<FormInput
  errorBefore={undefined}
  field={undefined}
  isForm={undefined}
  showErrors={undefined}>
  [Function]
</FormInput>
`;
